docs/shared/recipes/tips-n-tricks/switch-to-workspaces-project-references.md

```diff
@@ -0,0 +1,230 @@
+
+{% /tab %}
+{% tab label="yarn" %}
+
+```json {% fileName="package.json" %}
+{
+  "workspaces": ["apps/**", "libs/**"]
+}
+```
+
+Defining the `workspaces` property in the root `package.json` file lets yarn know to look for other `package.json` files in the specified folders. With this configuration in place, all the dependencies for the individual projects will be installed in the root `node_modules` folder when `yarn` is run in the root folder. Also, the projects themselves will be linked in the root `node_modules` folder to be accessed as if they were npm packages.
+
+{% /tab %}
+{% tab label="bun" %}
+
+```json {% fileName="package.json" %}
+{
+  "workspaces": ["apps/**", "libs/**"]
+}
+```
+
+Defining the `workspaces` property in the root `package.json` file lets bun know to look for other `package.json` files in the specified folders. With this configuration in place, all the dependencies for the individual projects will be installed in the root `node_modules` folder when `bun install` is run in the root folder. Also, the projects themselves will be linked in the root `node_modules` folder to be accessed as if they were npm packages.
+
+{% /tab %}
+{% tab label="pnpm" %}
+
+```yaml {% fileName="pnpm-workspace.yaml" %}
+packages:
+  # specify a package in a direct subdir of the root
+  - 'my-app'
+  # all packages in direct subdirs of packages/
+  - 'packages/*'
+  # all packages in subdirs of libs/
+  - 'libs/**'
+  # exclude packages that are inside test directories
+  - '!**/test/**'
+```
+
+Defining the `packages` property in the root `pnpm-workspaces.yaml` file lets pnpm know to look for project `package.json` files in the specified folders. With this configuration in place, all the dependencies for the individual projects will be installed in the root `node_modules` folder when `pnpm install` is run in the root folder.
+
+If you want to reference a local library project from an application, you need to include the library in the `devDependencies` of the application's `package.json` with `workspace:*` specified as the library's version. `workspace:*` tells pnpm that the project is in the same repository and not an npm package. You want to specify local projects as `devDependencies` instead of `dependencies` so that the library is not included twice in the production bundle of the application.
+
+```json {% fileName="/apps/my-app/package.json" %}
+{
+  "devDependencies": {
+    "@my-org/some-project": "workspace:*"
+  }
+}
+```
+
+{% /tab %}
+{% /tabs %}
+
+## Update Root TypeScript Configuration
+
+The root `tsconfig.base.json` should contain a `compilerOptions` property and no other properties. `compilerOptions.composite` and `compilerOptions.declaration` should be set to `true`. `compilerOptions.paths` should not be set.
```

Comment:
I believe the `rootDir` also needs to be removed as that now resides within each project-level `tsconfig.json` when switching to TS proj refs.